rex/renewable_resource.py

```diff
@@ -44,13 +51,16 @@ def get_SAM_df(self, site):
                                'Month': self.time_index.month,
                                'Day': self.time_index.day,
                                'Hour': self.time_index.hour})
-        if len(self) > 8784:
+
+        if len(self) > 8784 or (self.time_index.minute != 0).any():
             res_df['Minute'] = self.time_index.minute
 
         time_zone = self.meta.loc[site, 'timezone']
         time_interval = len(self.time_index) // 8760
 
-        for var in ['dni', 'dhi', 'wind_speed', 'air_temperature']:
+        main_cols = ['dni', 'dhi', 'wind_speed', 'air_temperature']
+        extra_cols = extra_cols or {}
```

Comment:
Interesting! I've never seen this. So it will go to the {} if extra_cols is bool(False)?

Reply:
Yup! I love using this pattern when I use an optional input with a default value of `None` in the function declaration. I feel it is an elegant way to get around the fact that you should never use mutables as default arguments.